src/util/http/HttpClient.h

```diff
@@ -58,13 +58,21 @@ class HttpClientImpl {
       std::string_view target, ad_utility::SharedCancellationHandle handle,
       std::string_view requestBody = "",
       std::string_view contentTypeHeader = "text/plain",
-      std::string_view acceptHeader = "text/plain");
+      std::string_view acceptHeader = "text/plain",
+      const std::unordered_map<std::string_view, std::string_view>&
+          customHeaders = {});
```

Comment:
That type is really odd and prone to dangling references.
Why not use `std::string` , as the types, HTTP business is expensive anyway.
And does a simple `std::vector<pair<string, string>>` also work? Header might be repeated in HTTP if I am not mistaken.